tests/rbd/rbd_features_disable.py

```diff
@@ -0,0 +1,97 @@
+from utility.log import Log
+from utility.utils import run_fio
+
+log = Log(__name__)
+
+
+def rbd_feature_disable(rbd, pool_type, **kw):
+    """
+    Disable the image features on image when image is moved to trash.
+    Args:
+        # feature_name: name of the feature to enable
+        rbd: rbd object
+        pool_type: pool type (ec_pool_config or rep_pool_config)
+        **kw: Test data
+    """
+    pool = kw["config"][pool_type]["pool"]
+    image = kw["config"][pool_type]["image"]
+    feature_name = "fast-diff"
+    try:
+        client = kw["ceph_cluster"].get_nodes(role="client")[0]
+        run_fio(image_name=image, pool_name=pool, client_node=client)
+
+        # Check if the feature is already enabled for the image
+        image_load = rbd.image_info(pool, image)
+        if feature_name in image_load.get("features", {}):
+            log.error(f"Feature {feature_name} is already enabled.")
+            pass
+
+        # Move the image to trash
+        if rbd.move_image_trash(pool, image):
+            log.error("RBD trash removal failed ")
+            return 1
+
+        # Try to Disable the feature on the image, should fail since image is in trash
+        try:
+            rbd.disable_rbd_feature(pool, image, feature_name)
+        except RbdBaseException as error:
+            log.info(
+                f"Expected error occurred: {error.message}  The image is not available for feature disable"
+            )
+
+        # Restore Image from trash and try to disable feature - should succeed
+        image_id = rbd.get_image_id(pool, image)
+        rbd.trash_restore(pool, image_id)
+        log.info("Image restored successfully from trash.")
+        if rbd.disable_rbd_feature(pool, image, feature_name):
+            log.error("RBD feature was not disabled successfully")
+            return 1
+
+        # Verify the feature is disabled and not present in the image info
+        if feature_name not in image_load.get("features", {}):
+            log.error(
+                f"Feature {feature_name} is present in the rbd info of {pool}/{image}."
+            )
+            return 1
+
+    except RbdBaseException as error:
+        log.error(error.message)
+        return 1
+
+
+def run(**kw):
+    """Enable the image features on image, and mark for deletion. Now try to  disable features.
+
+    Args:
+        kw: test data
+
+    Returns:
+        int: The return value. 0 for success, 1 otherwise
+
+    Pre-requisites :
+    We need atleast one client node with ceph-common package,
+    conf and keyring files
+
+    Test cases covered -
+    1) CEPH-11415 - Enable the image features on image, and mark for deletion. Now try to  disable features.
+    Test Case Flow
+    1. Create a pool and an Image - RBD features are enabled by default
+    2. generate IO in images
+    3. Move images to trash and perform disable rbd feature
+    4. Restore image from trash and verify that images is able to disable successfully
```

Comment:
Step 4 is looks something missing, i think it is image feature. 
same modify in PR description also.

Reply:
done